addons/project/models/project.py

```diff
@@ -115,11 +115,17 @@ def _compute_attached_docs_count(self):
             ])
 
     def _compute_task_count(self):
-        task_data = self.env['project.task'].read_group([('project_id', 'in', self.ids), '|', '&', ('stage_id.is_closed', '=', False), ('stage_id.fold', '=', False), ('stage_id', '=', False)], ['project_id'], ['project_id'])
+        task_data = self.env['project.task'].read_group([('informal_project_id', 'in', self.ids), '|', '&', ('stage_id.is_closed', '=', False), ('stage_id.fold', '=', False), ('stage_id', '=', False)], ['project_id'], ['project_id'])
         result = dict((data['project_id'][0], data['project_id_count']) for data in task_data)
         for project in self:
             project.task_count = result.get(project.id, 0)
 
+    def _compute_task_count_with_subtasks(self):
+        task_data = self.env['project.task'].read_group(['|', '&', ('stage_id.is_closed', '=', False), ('stage_id.fold', '=', False), ('stage_id', '=', False)], ['project_id'], ['project_id'])
```

Comment:
What do you think about making a single compute that set both fields (task_count and task_count_with_subtasks) by adding the informal_project_id in the group_by and by looping in python? By doing so we do not call twice the database. Another benefit of this is to keep the search_group domain at only one place with eases maintenance.

Reply:
I agree concerning maintenance ! :)

Since it's not used in the same views, I choose for this design at the first sight, and maybe would it be better to use a factorized/common method for the read_group.

But given your excellent idea, here is what I found, does that suite you ?

`python
    def _compute_task_count(self):
```suggestion
        task_data = self.env['project.task'].read_group(['|', '&', ('stage_id.is_closed', '=', False), ('stage_id.fold', '=', False), ('stage_id', '=', False)], ['project_id'], ['project_id'])
```
        task_data = self.env['project.task'].read_group(['|', '&', ('stage_id.is_closed', '=', False), ('stage_id.fold', '=', False), ('stage_id', '=', False)],
                                                        ['project_id', 'informal_project_id'], ['project_id', 'informal_project_id'])
        result_wo_subtask = defaultdict(int)
        result_with_subtasks = defaultdict(int)
        for data in task_data:
            result_wo_subtask[data['project_id'][0]] += data['informal_project_id_count']
            result_with_subtasks[data['project_id'][0]] += data['project_id_count']
        for project in self:
            project.task_count = result_wo_subtask.get(project.id, 0)
            project.task_count_with_subtasks = result_with_subtasks.get(project.id, 0)
            `
            
 ```suggestion
            def _compute_task_count(self):
        task_data = self.env['project.task'].read_group(['|', '&', ('stage_id.is_closed', '=', False), ('stage_id.fold', '=', False), ('stage_id', '=', False)],
                                                        ['project_id', 'informal_project_id'], ['project_id', 'informal_project_id'])
        result_wo_subtask = defaultdict(int)
        result_with_subtasks = defaultdict(int)
        for data in task_data:
            result_wo_subtask[data['project_id'][0]] += data['informal_project_id_count']
            result_with_subtasks[data['project_id'][0]] += data['project_id_count']
        for project in self:
            project.task_count = result_wo_subtask.get(project.id, 0)
            project.task_count_with_subtasks = result_with_subtasks.get(project.id, 0)
```